app/upgrades/v11/upgrades_test.go

```diff
@@ -51,44 +69,309 @@ func setupTestApp(t *testing.T) (*app.Evmos, sdk.Context) {
+		// Set Validator
+		valAddr, err := sdk.ValAddressFromBech32(a)
+		suite.Require().NoError(err)
+		validator, err := stakingtypes.NewValidator(valAddr, suite.consKey, stakingtypes.Description{})
+		suite.Require().NoError(err)
+		validator = stakingkeeper.TestingUpdateValidator(suite.app.StakingKeeper, suite.ctx, validator, true)
+		suite.app.StakingKeeper.AfterValidatorCreated(suite.ctx, validator.GetOperator())
+		err = suite.app.StakingKeeper.SetValidatorByConsAddr(suite.ctx, validator)
+		suite.Require().NoError(err)
+	}
+	validators := suite.app.StakingKeeper.GetValidators(suite.ctx, 1000)
+	suite.Require().Equal(len(validatorsAddr)+1, len(validators))
+}
+
+func (suite *UpgradeTestSuite) TestMigrateEscrowAcc() {
+	suite.SetupTest(evmostypes.MainnetChainID)
 
 	// fund some escrow accounts
 	existingAccounts := 30
-	setupEscrowAccounts(app, ctx, existingAccounts)
+	suite.setupEscrowAccounts(existingAccounts)
 
 	// Run migrations
-	v11.MigrateEscrowAccounts(ctx, app.AccountKeeper)
+	v11.MigrateEscrowAccounts(suite.ctx, suite.app.AccountKeeper)
 
 	// check account types for channels 0 to 36
 	for i := 0; i <= 36; i++ {
 		channelID := fmt.Sprintf("channel-%d", i)
 		addr := ibctypes.GetEscrowAddress(ibctypes.PortID, channelID)
-		acc := app.AccountKeeper.GetAccount(ctx, addr)
+		acc := suite.app.AccountKeeper.GetAccount(suite.ctx, addr)
 
 		if i > existingAccounts {
-			require.Nil(t, acc, "This account did not exist, it should not be migrated")
+			suite.Require().Nil(acc, "This account did not exist, it should not be migrated")
 			continue
 		}
-		require.NotNil(t, acc)
+		suite.Require().NotNil(acc)
 
 		moduleAcc, isModuleAccount := acc.(*authtypes.ModuleAccount)
-		require.True(t, isModuleAccount)
-		require.NoError(t, moduleAcc.Validate(), "account validation failed")
+		suite.Require().True(isModuleAccount)
+		suite.Require().NoError(moduleAcc.Validate(), "account validation failed")
+	}
+}
+
+func (suite *UpgradeTestSuite) TestDistributeRewards() {
+	communityPoolAccountAddress := sdk.MustAccAddressFromBech32("evmos1jv65s3grqf6v6jl3dp4t6c9t9rk99cd8974jnh")
+
+	balance, ok := sdk.NewIntFromString("7399998994000000000000000")
+	suite.Require().True(ok, "error converting rewards account balance")
+
+	expRewards, ok := sdk.NewIntFromString("5625000000302600000000000")
+	suite.Require().True(ok, "error converting rewards")
+
+	actualRewards := math.NewInt(0)
+	for _, currentElem := range v11.Allocations {
+		res, _ := sdk.NewIntFromString(currentElem[1])
+		actualRewards = actualRewards.Add(res)
+	}
+	suite.Require().Equal(expRewards, actualRewards)
+
+	var (
+		expCommPoolBalance = balance.Sub(expRewards)
+		noRewardAddr       = sdk.MustAccAddressFromBech32("evmos1009egsf8sk3puq3aynt8eymmcqnneezkkvceav")
+	)
+
+	testCases := []struct {
+		name            string
+		chainID         string
+		malleate        func()
+		expectedSuccess bool
+	}{
+		{
+			"Mainnet - success",
+			evmostypes.MainnetChainID + "-4",
```

Comment:
optional: In case we have same suffix number in all cases, would be more clean to define `chainID + "4"` variable once and reuse it